pkg/controller/external.go

```diff
@@ -199,18 +208,39 @@ func (e *external) Observe(ctx context.Context, mg xpresource.Managed) (managed.
 		return managed.ExternalObservation{}, errors.Wrap(err, "cannot set observation")
 	}
 
+	// NOTE(lsviben) although the annotations were supposed to be set and the
+	// managed resource updated during the Create step, we are checking and
+	// updating the annotations here due to the fact that in most cases, the
+	// Create step is done asynchronously and the managed resource is not
+	// updated with the annotations. That is why below we are prioritizing the
+	// annotations update before anything else. We are setting lateInitialized
+	// to true so that the reconciler updates the managed resource. This
+	// behavior conflicts with management policies in which LateInitialize is
+	// turned off. To circumvent this, we are checking if the management policy
+	// does not contain LateInitialize and if it does not, we are updating the
+	// annotations manually.
 	annotationsUpdated, err := resource.SetCriticalAnnotations(tr, e.config, tfstate, string(res.State.GetPrivateRaw()))
 	if err != nil {
 		return managed.ExternalObservation{}, errors.Wrap(err, "cannot set critical annotations")
 	}
+	policyHasLateInit := policySet.HasAny(xpv1.ManagementActionLateInitialize, xpv1.ManagementActionAll)
+	if annotationsUpdated && !policyHasLateInit {
+		if err := e.kube.Update(ctx, mg); err != nil {
```

Comment:
This may have performance implications because it may result in conflicts when/if the managed reconciler also decides to a spec update. 

I don't think this is super important because I assume the number of times `annotationsUpdated` will be `True` will probably be small but I'm 100% sure as it's not just the critical annotation keys but also the values. Just leaving a note here for future reference. What do you think?

Reply:
Similar to the answer [above](https://github.com/upbound/upjet/pull/224#discussion_r1269912443). It should only affect resources in which:
- managementPolicies are enabled
- `LateInitialization` is off

But good point to keep it in mind :+1: for possible performance implications later on.

Reply:
Yes, I believe why we initially relied on setting `LateInitialized: True` in the first place (as you mentioned above) is to prevent conflicts with the managed reconciler's updates by trying to have the managed reconciler as the sole entity doing these updates. Let's continue with this implementation and we will need to act if we observe any adverse affects.